plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -5314,6 +5314,24 @@ public void testInsertingIntoTablesWithColumnsWithQuotesInName()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testInsertIntoBucketedColumnWhenTaskWriterCountIsGreaterThanOrEqualToNodeCount()
+    {
+        int taskWriterCount = 4;
+        assertThat(taskWriterCount).isGreaterThanOrEqualTo(getQueryRunner().getNodeCount());
+        Session session = Session.builder(getSession())
+                .setSystemProperty("task_writer_count", String.valueOf(taskWriterCount))
+                .build();
+
+        String tableName = "test_inserting_into_bucketed_column_when_task_writer_count_is_greater_than_or_equal_to_node_count_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (bucketed_col INT) WITH (partitioning = ARRAY['bucket(bucketed_col, 10)'])");
+
+        assertUpdate(session, "INSERT INTO " + tableName + " VALUES (1)", 1);
+        assertQuery("SELECT * FROM " + tableName, "VALUES 1");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
```

Comment:
BTW the problem doesn't look Iceberg specific. Should this be part of BCT?

Reply:
Do we have a way of creating bucketed tables in BCT? I think only Hive and Iceberg would support this.

Reply:
Is the problem about bucketed tables only? Are "plainly partitioned" tables not affected?

Anyway, i hear you on the test setup challenge, making it hard to test in BCT.